rosbag2_transport/src/rosbag2_transport/recorder.cpp

```diff
@@ -523,6 +523,20 @@ std::shared_ptr<rclcpp::GenericSubscription>
 RecorderImpl::create_subscription(
   const std::string & topic_name, const std::string & topic_type, const rclcpp::QoS & qos)
 {
+#if defined _WIN32 || defined __CYGWIN__
+  return node->create_generic_subscription(
+    topic_name,
+    topic_type,
+    qos,
+    [this, topic_name, topic_type](std::shared_ptr<const rclcpp::SerializedMessage> message,
+    const rclcpp::MessageInfo & mi) {
+      if (!paused_.load()) {
+        writer_->write(
+          std::move(message), topic_name, topic_type, node->now().nanoseconds(),
+          node->now().nanoseconds());
```

Comment:
According to the MCAP spec, the source timestamp shall be assigned to **0** if not available.